src/main/java/life/qbic/io/commandline/PostmanCommandLineOptions.java

```diff
@@ -1,85 +1,161 @@
-  public Path filePath;
-
-  @Option(
-      names = {"-c", "--conserve"},
-      description = "Conserve the file path structure during download")
-  public boolean conservePath;
-
   @Option(
-      names = {"--print"},
-      description = "print available datasets for the provided samples")
-  public boolean printDatasets;
+          names = {"-as", "-as_url"},
+          description = "ApplicationServer URL",
+          scope = CommandLine.ScopeType.INHERIT)
+  public String as_url = "https://qbis.qbic.uni-tuebingen.de/openbis/openbis";
 
   @Option(
-      names = {"-b", "--buffer-size"},
-      description =
-          "dataset download performance can be improved by increasing this value with a multiple of 1024 (default)."
-              + " Only change this if you know what you are doing.")
-  public int bufferMultiplier = 1;
+          names = {"-dss", "--dss_url"},
+          description = "DataStoreServer URL",
+          scope = CommandLine.ScopeType.INHERIT)
+  public String dss_url = "https://qbis.qbic.uni-tuebingen.de/datastore_server";
 
   @Option(
-      names = {"-s", "--suffix"},
-      description = "returns all files of datasets containing the supplied suffix")
-  public List<String> suffixes = new ArrayList<>();
+          names = {"-f", "--file"},
+          description = "a file with line-separated list of QBiC sample ids",
+          scope = CommandLine.ScopeType.INHERIT)
+  public Path filePath;
 
   @Option(
-      names = {"-r", "--regex"},
-      description = "returns all files of datasets using your supplied regular expression")
-  public List<String> regexPatterns = new ArrayList<>();
+          names = {"-h", "--help"},
+          usageHelp = true,
+          description = "display a help message and exit",
+          scope = CommandLine.ScopeType.INHERIT)
+  public boolean helpRequested = false;
 
+  // not used currently
   @Option(
       names = {"-t", "--type"},
-      description = "filter for a given openBIS dataset type")
+      description = "filter for a given openBIS dataset type",
+      scope = CommandLine.ScopeType.INHERIT)
   public String datasetType = "";
 
-  @Option(
-      names = {"-dss", "--dss_url"},
-      description = "DataStoreServer URL")
-  public String dss_url = "https://qbis.qbic.uni-tuebingen.de/datastore_server";
+  private List<String> verifyProvidedIdentifiers() throws IOException {
+    if ((ids == null || ids.isEmpty()) && filePath == null) {
+      System.out.println(
+          "You have to provide one ID as command line argument or a file containing IDs.");
+      System.exit(1);
+    } else if ((ids != null) && (filePath != null)) {
+      System.out.println(
+          "Arguments --identifier and --file are mutually exclusive, please provide only one.");
+      System.exit(1);
+    } else if (filePath != null) {
+      ids = IdentifierParser.readProvidedIdentifiers(filePath.toFile());
+    }
+    return ids;
```

Comment:
Here you specify, that the file has precedence over the ids. Is this related to the issue you mentioned?

Reply:
I don't think that the file has precedence over the ids. They are mutually exclusive. The first if case in this method adresses the case, that no ids or file is given. The second one is for the case, that both is given. This is not possible and there will be an error message. Therefore, the third if case handles the case that only a file is given. Because we need the ids in the file to proceed, the readProvidedIdentifiers functions is called here. The result is set to a String list ids, but this list did not exist in this case before and is therefore not overwritten.

The issue I mentioned only occures when I tried to handle this "mutually exclusive options case" with the option @ArgGroup that is provided by picocli. It should be possible to define groups that are mutually exclusive. But because it did not work like this and I also found a similar issue on the github page of picocli which also did not have a solution yet, I kept the functionality that handled the issue before in the CommandLineParser class in the function verifyProvidedIdentifiers().